src/formulas/compiler.ts

```diff
@@ -211,139 +178,101 @@ export function compile(formula: string): CompiledFormula {
             }
           }
 
-          const compiledAST = compileAST(currentArg, isLazy, isMeta, hasRange, {
+          const compiledAST = compileAST(currentArg, isMeta, hasRange, {
             functionName: ast.value.toUpperCase(),
             paramIndex: i + 1,
           });
-          listArgs.push(compiledAST);
+          compiledArgs.push(isLazy ? compiledAST.wrapInClosure() : compiledAST);
         }
       }
 
-      return listArgs;
+      return compiledArgs;
     }
 
     /**
      * This function compiles all the information extracted by the parser into an
      * executable code for the evaluation of the cells content. It uses a cash to
      * not reevaluate identical code structures.
      *
-     * The function is sensitive to two parameters “isLazy” and “isMeta”. These
-     * parameters may vary when compiling function arguments:
-     *
-     * - isLazy: In some cases the function arguments does not need to be
-     * evaluated before entering the functions. For example the IF function might
-     * take invalid arguments that do not need to be evaluate and thus should not
-     * create an error. For this we have lazy arguments.
-     *
-     * - isMeta: In some cases the function arguments expects information on the
+     * The function is sensitive to parameter “isMeta”. This
+     * parameter may vary when compiling function arguments:
+     * isMeta: In some cases the function arguments expects information on the
      * cell/range other than the associated value(s). For example the COLUMN
      * function needs to receive as argument the coordinates of a cell rather
      * than its value. For this we have meta arguments.
      */
-
     function compileAST(
       ast: AST,
-      isLazy = false,
       isMeta = false,
       hasRange = false,
       referenceVerification: {
         functionName?: string;
         paramIndex?: number;
       } = {}
-    ): CompiledAST {
-      const codeBlocks: string[] = [];
-      let id, fnName, statement;
+    ): FunctionCode {
+      const code = new FunctionCodeBuilder(scope);
```

Comment:
function ? functionCode ?
```suggestion
      const function = new FunctionCodeBuilder(scope);
```